### PR TITLE
fixes for tunneldigger

### PR DESCRIPTION
**Pull request by nicolasberens**
the newly set-up d.tunnel is now using a minimal debian 12.

Therefore i added it to the "propper" hosts (e.g. remove expections)
I also added needed dependencies and tadded smaller fixes for hosts with Predictable Network Interface renaming

### CODE REVIEW

File: play.yml [hunk 1]
```diff
@@ -51,7 +52,9 @@
     - ff_monitor
 
 - name: Set up tunneldigger
-  hosts: c.tunnel.berlin.freifunk.net,f.tunnel.berlin.freifunk.net
+  hosts: c.tunnel.berlin.freifunk.net,f.tunnel.berlin.freifunk.net,d.tunnel.berlin.freifunk.net
```
Comment (pmelange) [inline play.yml:55]: Can you please put the hosts in alphabetical order?

Is there some kind of annotation such as [c|d|f].tunnel.berlin.freifunk.net?

File: roles/tunneldigger/defaults/main.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ---
 tunneldigger_path: /opt/tunneldigger
-tunneldigger_listen_ip: "{{ ansible_eth0.ipv4.address }}"
+tunneldigger_listen_ip: "{{ ansible_default_ipv4.address }}"
 tunneldigger_snat_ip: "{{ tunneldigger_listen_ip }}"
 tunneldigger_oifname: "eth0"
```
Comment (pmelange) [inline roles/tunneldigger/defaults/main.yml:5]: Here is still eth0.  Should it be ansible_default_ipv4.interface?

File: roles/tunneldigger/templates/nftables.conf.j2 [hunk 3]
```diff
@@ -6,16 +6,16 @@ flush ruleset
 table ip nat {
         chain postrouting {
                 type nat hook postrouting priority srcnat; policy accept;
-                ip saddr 172.31.224.0/20 oif "eth0" snat to {{ tunneldigger_snat_ip }}
+                ip saddr 172.31.224.0/20 oif "{{ ansible_default_ipv4.interface }}" snat to {{ tunneldigger_snat_ip }}
```
Comment (pmelange) [inline roles/tunneldigger/templates/nftables.conf.j2:9]: If the host has a range of IP addresses, the snat rule should look like

```ip saddr 172.31.224.0/20 oif "{{ ansible_default_ipv4.interface }}" snat to $STARTIP-$ENDIP```

But only if there are multiple IP addresses.  

When there is just one IP, then $STARTIP-§ENDIP would look like 192.168.0.1-192.168.0.1 (example) which is invalid.

File: roles/tunneldigger/defaults/main.yml (hunk 2 above)
Comment (pmelange) [reply roles/tunneldigger/defaults/main.yml:5]: Actually, it looks like ```tunneldigger_oifname``` is not used anywhere.  This line could also be removed

**Review by Akira25 — APPROVED** [review]
Besides the Things mentiooned by pmelange already, this looks good to me. Would be great, If you can fix them

File: roles/tunneldigger/templates/nftables.conf.j2 (hunk 3 above)
Comment (nicolasberens) [reply roles/tunneldigger/templates/nftables.conf.j2:9]: we currently don't really use this, but in theory we can just set `tunneldigger_snat_ip` for the host that has multiple to e.g. "192.168.0.1-192.168.0.10"

Comment (pmelange) [reply roles/tunneldigger/templates/nftables.conf.j2:9]: Nice!

**Review by pmelange — APPROVED** [review]

**Pull request merged**